install_base_ubuntu.sh

```diff
@@ -0,0 +1,128 @@
+
+#=== FUNCTION ========================================================
+# NAME: check root permision
+# DESCRIPTIN: Print confirmation. If not root, switch to root user
+#=====================================================================
+check_root()
+{
+    printf "Checking root permision...\n"
+
+    local usr=`whoami`
+
+    # checking whether current user is root or not
+    if [ ! "$usr" == "root" ];
+    then
+        printf "No root permision to install the package. Switching to root.\n"
+        sudo su
+    else
+        printf "Current user is: $usr"
+    fi
+}
+
+#=== FUNCTION ========================================================
+# NAME: install_pkg
+# DESCRIPTIN: function to install the package
+# PARAMETER 1: package name
+#=====================================================================
+install_pkg()
+{
+    local pkg=$1
+    printf "@@Installing package $pkg ... \n"
+
+    # with -y option to answer yes to user prompt when installing a package
+    apt-get -y install $pkg
+}
+
+#=== FUNCTION ========================================================
+# NAME: remove_pkg
+# DESCRIPTIN: function to remove the package
+# PARAMETER 1: package name
+#=====================================================================
+remove_pkg()
+{
+    local pkg=$1
+    printf "@@Removing package $pkg ... \n"
+
+    # with -y option to answer yes to user prompt when removing a package
+    apt-get -y remove $pkg
+}
+
+#=== FUNCTION ========================================================
+# NAME: pip_install
+# DESCRIPTIN: install Python packages using pip
+# PARAMETER 1: package name
+#=====================================================================
+pip_install()
+{
+    local pkg=$1
+    printf "@@pip installing package $pkg ... \n"
+
+    # non-interactive install using pip
+    yes w | pip install $pkg
+}
+#=== FUNCTION ========================================================
+# NAME: main
+# DESCRIPTIN: main function
+#=====================================================================
+main()
+{
+
+    # initialise listis of pre-requisite packages, ones that need to be removed and pip packages
+    # add more package names here, separated by space
+    pre_requisite_pkgs="build-essential autoconf automake libtool pkg-config trace-cmd sshpass \
+        kernelshark nmap net-tools tree python-matplotlib python-numpy libfreetype6-dev \
+        libpng12-dev python-nose python-pip python-dev iputils-ping git wget"
+
+    # packages that need to be remove
+    # add more package names here, separated by space
+    remove_pkgs="ipython ipython-notebook"
+
+    # packages that need to be installed by pip
+    # add more package names here, separated by space
+    pip_pkgs="Cython trappy bart-py devlib psutil wrapt ipython jupyter"
+
+
+    # check root permision
+    # No need to do this on LAVAv2 LXC container
+    # check_root
+
+    # remove some packages
+    for package in $remove_pkgs
```

Comment:
Are not these loops a bit overkilling?

Why not just a simple:

```bash
sudo apt-get -y remove $remove_pkgs
sudo apt-get -y install $pre_requisite_pkgs
```

The same for pip installed packages...